cEP-0010.md

```diff
@@ -150,6 +163,10 @@ Here is the list of aspects to be implemented along with their corresponding
   `Root.Formatting.FileLength.max_file_length`.
 - `Root.Formatting.FileLength.max_file_length`: This taste represents the
   maximum length for a file.
+- `Root.Formatting.Length.FileLength.min_file_length`: This taste represents the
+  minimum number of lines per file.
+- `Root.Formatting.Length.FileLength.include_blank_lines`: This taste checks
```

Comment:
`include_` and `exclude_` should be avoided in a taste name, as they presume a default value which is unstated, or are relative to some value which is unstated.

I think this could be adjusted to `count_blank_lines`

But maybe a better approach would be to define a taste which allows the user to choose one of the various (code) line counting algorithms which are commonly used, and add an algorithm for the very dumb "all lines except blank lines" approach.

Reply:
It's a great idea but currently, bear settings follow `exclude_blank_lines` settings only and not such algorithms so we won't be able to map such algorithms to bear settings. To provide such algorithms, explicit implementation can be done in the bear. 

I think common algorithms include only `count_blank_lines` and `count_comment_lines` (tell me if I am missing something or misguided) and we can provide separate tastes for both.

But if you still feel that counting algorithms are necessary and things like `number of blank lines` and `number of comment lines` should be displayed, we can use cloc (https://github.com/AlDanial/cloc) instead of explicitly writing our own algorithms. But again this requires lot of implementation in the bear settings also.

IMO separate tastes for the separate algorithms can be made, instead of just one taste, allowing the user to choose more than one algorithm. After all, the user can choose both `count_blank_lines=True` and  `count_comment_lines=True` which counts lines including both blanks and comments and not just only one.

Let me know which design sounds better to you so that I can proceed?

Reply:
One more thing, I don't think `count_blank_lines` should be used instead of `include_blank_lines` as it would suggest counting of blank lines only and not including or excluding blank lines while counting.

Reply:
@pareksha , aspects have a spec, which describes how they work, and how they should be named.
https://github.com/coala/cEPs/blob/master/cEP-0005.md#naming-of-aspects-and-tastes

`include_blank_lines` is out.  Obviously not acceptable per the spec.  I do appreciate your opposition to `count_blank_lines`; when I look it now, it isnt much better per spec, and it is also a bit ambiguous as you point out.
It is your task to find appropriate names, not my job.

We do not name aspects/tastes to work with existing bear settings.
We design the aspects according to the spec, and design them too work for all languages that are relevant, and then *implement* them in the bears.  That may involve some bear settings being impossible to match to aspects, and that is ok.  Sometimes the bear needs to be redesigned; other times the bear setting can be mapped to an aspect/taste via a complex mapping; other times the bear setting is correctly designed, but just isnt able to be mapped, and the user will need to play with the setting directly.

There are *lots* of papers about measuring loc, and all the odd cases that need to be considered in such an algorithm.  Again, it is part of your project to analyse this.  Start at https://en.wikipedia.org/wiki/Source_lines_of_code .  Also find some SLOC software and read it to see some of the oddities it needs to deal with.  Do that with several programming languages.

Reply:
I read about the naming conventions and surely I have a better understanding of taste and aspect names now. I totally agree that `include_blank_lines` should be out simply because itself it does not deliver the total information. It somewhat seems to be incomplete and less informative. Plus `count_blank_lines` is ambiguous. 

IMO there's no good in debating about this now as we are already thinking of dropping this taste and putting a taste (or more) for separate algorithms for counting SLOC.

I will be investigating about SLOC softwares and will update on that soon.